gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/jupiter/core/context/OnMessagesInterceptor.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.core.context;
+
+import io.gravitee.gateway.jupiter.api.message.Message;
+import io.reactivex.FlowableTransformer;
+import java.util.function.Function;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public interface OnMessagesInterceptor {
+    /**
+     * Set an interceptor allowing to capture any call to {@link MutableRequest#onMessages(FlowableTransformer)} or {@link MutableResponse#onMessages(FlowableTransformer)}.
+     * The captured {@link FlowableTransformer} will then be provided to the specified function which will be able to provide its own {@link FlowableTransformer}.
+     * <p/>
+     * This can be particularly useful to avoid, replace or compose the transformation set by a policy on the incoming messages.
+     * <p/>
+     * Ex: calculate the message content size before and after a policy is applied on a message.
+     * <p/>
+     * <pre>
+     *      request.setOnMessagesInterceptor(onMessages ->
+     *             messages -> {
+     *                 final AtomicLong initialTotalSize = new AtomicLong(0);
+     *                 final AtomicLong finalTotalSize = new AtomicLong(0);
+     *                 return messages
+     *                     .doOnNext(message -> initialTotalSize.addAndGet(message.content().length()))
+     *                     .compose(onMessages)
+     *                     .doOnNext(message -> finalTotalSize.addAndGet(message.content().length()));
+     *             }
+     *         );
+     * </pre>
+     *
+     * @param interceptor the function taking the {@link FlowableTransformer} provided by the call to {@link MutableRequest#onMessages(FlowableTransformer)} or {@link MutableResponse#onMessages(FlowableTransformer)} and returns another {@link FlowableTransformer} as a replacement.
+     */
+    void setOnMessagesInterceptor(Function<FlowableTransformer<Message, Message>, FlowableTransformer<Message, Message>> interceptor);
```

Comment:
There is no requirement for now. I prefer to wait for a concrete need and not anticipate too much (interceptor is core concept so I suppose it will be pretty easy to change). Ok for you ?

Reply:
ok fair enough.